xbmc/rendering/dx/ScreenshotSurfaceWindows.cpp

```diff
@@ -0,0 +1,100 @@
+/*
+ *  Copyright (C) 2005-2018 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "ScreenshotSurfaceWindows.h"
+
+#include "guilib/GUIComponent.h"
+#include "guilib/GUIWindowManager.h"
+#include "rendering/dx/DeviceResources.h"
+#include "ServiceBroker.h"
+#include "threads/SingleLock.h"
+#include "utils/log.h"
+#include "windowing/GraphicContext.h"
+
+#include "system_gl.h"
+
+#include <wrl/client.h>
+
+using namespace Microsoft::WRL;
+
+bool CScreenshotSurfaceWindows::Capture()
+{
+  CWinSystemBase *winsystem = CServiceBroker::GetWinSystem();
+  if (!winsystem)
+    return false;
+
+  CGUIComponent *gui = CServiceBroker::GetGUI();
+  if (!gui)
+    return false;
+
+  CSingleLock lock(winsystem->GetGfxContext());
+  gui->GetWindowManager().Render();
+
+  auto deviceResources = DX::DeviceResources::Get();
+  deviceResources->FinishCommandList();
+
+  ComPtr<ID3D11DeviceContext> pImdContext = deviceResources->GetImmediateContext();
+  ComPtr<ID3D11Device> pDevice = deviceResources->GetD3DDevice();
+  CD3DTexture& backbuffer = deviceResources->GetBackBuffer();
```

Comment:
Can this be a `const &`?

Reply:
I don't know, the code is just copied from before. I don't mess with windows